pkg/custom-plugins/plugin.go

```diff
@@ -0,0 +1,88 @@
+package customplugins
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+// Plugin represents a plugin spec.
+type Plugin struct {
+	// Steps is a sequence of steps to run for this plugin.
+	// The steps are executed in order.
+	// If a step fails, the execution stops and the error is returned.
+	Steps []PluginStep `json:"steps,omitempty"`
+}
+
+// Validate validates all the plugin steps.
+func (p Plugin) Validate() error {
```

Comment:
Let's use pointer receiver first.

```suggestion
func (p *Plugin) Validate() error {
```